spec/api/allocation_api_spec.rb

```diff
@@ -5,6 +5,10 @@
 describe 'Allocation API', vcr: { cassette_name: 'prison_api/allocation_api' } do
   let(:Authorization) { "Bearer TEST_TOKEN" }
 
+  before do
+    allow_any_instance_of(DomainEvents::Event).to receive(:publish).and_return(nil)
+  end
```

Comment:
Should a controller spec be mocking out a dependency of the model it uses?

Or should it be mocking out the whole model itself?

I thought mocks were meant to be used to mock out the dependencies of the class being tested, not the dependencies of the dependencies of the class being tested?

I ask because we already have so much trouble with tests that mock things out 3-4 levels deep, and we have to understand 2-3 layers of the code just to understand and fix our tests, was hoping we didn't make the problem worse.

I noticed you're having to do this for many controllers simply because that's the way the code's been architected already, so I guess it's a consequence of the existing tests being a pile of shit. If that's the case, then please proceed and ignore my comment.